src/test/java/teammates/logic/core/NotificationsLogicTest.java

```diff
@@ -55,14 +63,43 @@ private void testGetNotification() throws Exception {
         assertNull(notifLogic.getNotification("nonexistant-notification"));
     }
 
+    private void testGetAllNotifications() throws Exception {
+        ______TS("success: retrieve all when there exist notifications");
+
+        n = getTypicalNotification();
+        notifDb.createEntity(n);
+        NotificationAttributes n1 = getTypicalNotification();
+        n1.setNotificationId("new-notification-id");
+        notifDb.createEntity(n1);
+
+        List<NotificationAttributes> retreivedNotifs = notifLogic.getAllNotifications();
+        retreivedNotifs.forEach(this::verifyPresentInDatabase);
+
+        removeNotificationsFromDb(n, n1);
+    }
+
+    private void testGetActiveNotificationsByTargetUser() throws Exception {
+        ______TS("success : valid target user");
+
+        n = getTypicalNotification();
+        n.setTargetUser(NotificationTargetUser.STUDENT);
+        notifDb.createEntity(n);
+        List<NotificationAttributes> retreivedNotifs =
+                notifLogic.getActiveNotificationsByTargetUser(NotificationTargetUser.STUDENT);
+
+        assertTrue(retreivedNotifs.contains(n));
```

Comment:
Would you think that this is not sufficient? You may want to a few more to DB (including expected to be retrieved and those that are not).

You may want to take a look at `NotificationsDbTest` - It basically loads all typical data into DB, then execute this function and check whether the returned ones are the expected ones.

Reply:
Okay sure, thanks for the suggestion!